src/cf-tables/src/TableSortable.js

```diff
@@ -2,211 +2,215 @@
 
+const TableSortable = {
   ui: {
     base:       '.o-table__sortable',
     tableBody:  'tbody',
     sortButton: '.sorted-up, .sorted-down'
   },
 
-  /**
-   * Function used to create computed and triggered properties.
-   */
-  initialize: function() {
-    this.sortClass = UNDEFINED;
-    this.sortColumnIndex = UNDEFINED;
-    this.sortDirection = UNDEFINED;
-    this.tableData = [];
-    this.bindProperties();
-    if ( this.ui.sortButton ) {
-      this.sortColumnIndex = this.getColumnIndex();
-      this.sortDirection =
-        this.contains( this.ui.sortButton, this.classes.sortDown ) ?
-        DIRECTIONS.DOWN : DIRECTIONS.UP;
-      this.updateTable();
-    }
-  },
-
-  /**
-   * Function used to create computed and trigger properties.
-   */
-  bindProperties: function() {
-    var sortDirection;
-
-    Object.defineProperty( this, 'sortDirection', {
-      configurable: true,
-      get: function() {
-        return sortDirection;
-      },
-      set: function( value ) {
-        if ( value === DIRECTIONS.UP ) {
-          this.sortClass = this.classes.sortUp;
-        } else if ( value === DIRECTIONS.DOWN ) {
-          this.sortClass = this.classes.sortDown;
-        }
-        sortDirection = value;
-      }
-    } );
+  classes: {
+    sortDown: 'sorted-down',
+    sortUp:   'sorted-up'
   },
 
-  /**
-   * Function used to get the column index of the active sort column.
-   *
-   * @param {HTMLNode} element - The element used as the sortable.
-   * @returns {number} The column index of the active sort column.
-   */
-  getColumnIndex: function( element ) {
-    return closest( element || this.ui.sortButton, 'td, th' ).cellIndex;
+  events: {
+    'click .sortable': 'onSortableClick'
   },
 
-  /**
-   * Function used to update the table data and dom.
-   * @returns {boolean} TODO: Add description.
-   */
-  updateTable: function() {
-    return this.updateTableData() && this.updateTableDom();
-  },
+  initialize:      initialize,
+  bindProperties:  bindProperties,
+  getColumnIndex:  getColumnIndex,
+  updateTable:     updateTable,
+  updateTableData: updateTableData,
+  updateTableDom:  updateTableDom,
+  tableDataSorter: tableDataSorter,
+  onSortableClick: onSortableClick
+};
 
-  /**
-   * Function used to get, sort, and update the table data array.
-   *
-   * @param {number} columnIndex - The index of the column used for sorting.
-   * @returns {Array} TODO: Add description.
-   */
-  updateTableData: function( columnIndex ) {
-    var cell;
-    var rows = this.ui.tableBody.querySelectorAll( 'tr' );
-    var sortType;
-    this.tableData = [];
-    columnIndex = columnIndex || this.sortColumnIndex;
-
-    for ( var i = 0, len = rows.length; i < len; ++i ) {
-      cell = rows[i].cells[columnIndex];
-      if ( cell ) {
-        cell = cell.textContent.trim();
+/**
+ * Function used to create computed and triggered properties.
+ */
+function initialize() {
+  this.sortClass = UNDEFINED;
```

Comment:
Should these be part of a `customClasses` object like in the Expandables code instead of properties?

Reply:
This looks like a different use case. In Expandables there's a set of class names that are assigned to constants that are only used inside the `initialize` method, so their scope is fine. In this case, `sortClass` is not constant and is used across methods, so is assigned as a property to share between methods. Personally I prefer not to expose anything publicly that is not needed in the public API, lest it leads to unintended implementations, so if `sortClass` could be internal that would be better (such as by moving it's location above the scope of the TableSortable object), but I don't think that's currently possible here as the methods here get added to instances of the Table organism, so that if `sortClass` was in a higher scope it would then become shared among all Table organisms using the TableSortable modifier. Is that right @sebworks?

Reply:
yep, that is correct @anselmbradford

Reply:
Hmm, sidenote that my renaming these capitalized like they are classes when they're not instantiating new instances is possibly misleading.